src/search/engines/_sedefaults/requests.go

```diff
@@ -9,24 +9,28 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
-func DoGetRequest(urll string, anonurll string, colCtx *colly.Context, collector *colly.Collector, packageName engines.Name, retError *error) {
+func DoGetRequest(urll string, anonurll string, colCtx *colly.Context, collector *colly.Collector, packageName engines.Name, errChannel chan error) {
 	log.Trace().
 		Str("engine", packageName.String()).
 		Str("url", anonurll).
 		Msg("GET")
 	err := collector.Request("GET", urll, nil, colCtx, nil)
 	if err != nil {
-		*retError = fmt.Errorf("%v.Search(): failed GET request to %v with %w", packageName.ToLower(), urll, err)
+		errChannel <- fmt.Errorf("%v.Search(): failed GET request to %v with %w", packageName.ToLower(), urll, err)
+	} else {
+		errChannel <- nil
```

Comment:
Why even push `nil` into the channel?

Reply:
Because we need same number of responses into the channel as there are requests made.